pkg/cloudmap/resource_manager.go

```diff
@@ -288,6 +296,49 @@ func (m *defaultResourceManager) createCloudMapService(ctx context.Context, vn *
 	}
 }
 
+func (m *defaultResourceManager) updateCloudMapService(ctx context.Context, svcSummary *serviceSummary, vn *appmesh.VirtualNode, nsSummary *servicediscovery.NamespaceSummary, serviceName string,
+	cloudMapTTL int64) (*serviceSummary, error) {
+	if awssdk.StringValue(nsSummary.Type) == servicediscovery.NamespaceTypeDnsPrivate {
+		actualsvcSummary := *svcSummary
+		desiredsvcSummary := BuildCloudMapServiceSummary(ctx, svcSummary, cloudMapTTL)
+		opts := cmp.Options{
+			cmpopts.EquateEmpty(),
+			cmp.AllowUnexported(serviceSummary{}),
+		}
+		// return if no change to service summary
+		if cmp.Equal(*desiredsvcSummary, actualsvcSummary, opts) {
+			return svcSummary, nil
+		}
+		diff := cmp.Diff(*desiredsvcSummary, actualsvcSummary, opts)
+		m.log.V(1).Info("cloudMap service changed",
+			"actualDnsConfig", actualsvcSummary,
+			"desiredDnsConfig", *desiredsvcSummary,
+			"diff", diff,
+		)
+		m.log.V(1).Info("Sending cloudMap service update request")
+		operationId, err := m.updateCloudMapServiceUnderPrivateDNSNamespace(ctx, vn, nsSummary, serviceName, cloudMapTTL, svcSummary)
+		if err != nil {
+			return nil, err
+		}
+		// wait for update to succeed or timeout and udpate on next reconcile loop
+		operationInput := &servicediscovery.GetOperationInput{
+			OperationId: operationId,
+		}
+		status := AwaitOperationSuccess(30, 1, func() string {
```

Comment:
I don't understand why need this logic at all. The aws sdk for go has retry and timeout options at the client level. We should be configuring these, not adding our own logic for this. (see https://aws.github.io/aws-sdk-go-v2/docs/configuring-sdk/retries-timeouts/)

Reply:
Just to clarify after my conversation with @melaniesifen. My question centered around why we need AwaitOperationSuccess when the SDK provides retries and timeouts. It turns out that Cloudmap has an async API: You do an operation, you get a request id back and you poll (via a different API) for the status of that request. Hence why @melaniesifen added that function. Which now makes sense. My recommendation to her is that this is a special purpose function specifically designed to poll for the Cloudmap status change (from PENDING to SUCCESS or FAIL). So, might want to either move the call to cloudMapSDK.GetOperation inside the function or change the prototype of the lambda to take the appropriate enumerated return type. Also, we might want to change the name of the function to better reflect its purpose... Something like PollCloudmapStatus ... etc. Lastly, the interior of the function doesn't need to have a go routine since the call to Await... is blocking. i.e. you are sitting in a loop polling for status until you are out of time. The previous implementation (with 2 go routines, one for timeout and one for the actual blocking REST call) was tricky because you could timeout and exit the main function but the poll routine might keep running or be blocked. Generally, we don't want to leave orphaned and blocked go routines or threads just hanging around if we can help it. Which brings me to my last point: lets still understand timeouts and retries for the HTTP Client. If the HTTP client has a default timeout of 30 seconds and a retry count of 3 etc., we might be out to lunch for 60 seconds when we called AwaitOperationSuccess with an expected timeout of 30 seconds (for example). I don't we can completely avoid something like this, but we should reduce the timeout overshoot if we can help it.